app/models/iiif_harvester.rb

```diff
@@ -0,0 +1,43 @@
+##
+# Harvester used for retrieving IIIF manifests from Vatican resources
+class IiifHarvester
+  attr_reader :iiif_manifest_url
+  ##
+  # @param [String] iiif_manifest_url
+  def initialize(iiif_manifest_url)
+    @iiif_manifest_url = iiif_manifest_url
+  end
+
+  def id
+    manifest['@id']
+  end
+
+  def shelfmark
+    id
+      .sub('https://digi.vatlib.it/iiif/', '')
+      .sub('/manifest.json', '')
+  end
+
+  def slug
+    shelfmark
+      .tr('.', '_') # TODO: Figure out what we want our ids to be
```

Comment:
Could we ticket this and pop the issue number in this TODO?

Reply:
Sure thing.. ticketed this morning in https://github.com/sul-dlss/vatican_exhibits/issues/37 and this code was copied from its previous location.